### DISCUSSION TITLE
Why Subscribing to 100 channles use 100 connections?

### DISCUSSION THREAD

**Discussion by strong84** [redis/go-redis#2499] [Q&A]
When I do:

```
rdb = redis.NewClient(&redis.Options{
		PoolSize:    1,
		Addr:        ip + ":" + port,
		Password:    pwd,
		DB:          0, // use default DB
		MaxRetries:  2,
		DialTimeout: 2 * time.Second,
	})

	for i := 0; i < 100; i++ {
		rdb.Subscribe(ctx, fmt.Sprintf("S%d", i))
		log.Write(fmt.Sprintf("S%d", i))
	}
```
and then check on redis server:
`CLIENT LIST`
I can see 100 connections are created.

and
`INFO CLIENTS` shows **connected_clients:100+**

Should it be just 1 connection from single rdb client instance?

**Comment by strong84** [comment]
`redigo` seems can subscribe multiple channels with just 1 connection.
I don't know where I understand wrong.

**Comment by monkey92t** [comment]
`Subscribe(ctx context.Context, channels ...string)` API creates a PubSub client object that allows subscribing to multiple channels. 

Similar to the `redis.NewClient()` API, we should create a`*redis.Client` object to execute many commands instead of creating a `*redis.Client` object for each command execution.

```go
pubsub := rdb.Subscribe(ctx, "chan_one", "chan_two"...)
...
```

:)

**Reply by monkey92t** [reply]
If you need to dynamically increase or decrease subscriptions, you can do so by:

```go
pubsub := rdb.Subscribe(ctx, "c1")
.....
.....
// add sub c2 c3 c4
pubsub.Subscribe(ctx, "c2", "c3", "c4")

// unsubscribe c1
pubsub.Unsubscribe(ctx, "c1")
```

**Reply by strong84** [reply]
Awesome~ Thanks for the explanation.
